src/nm.c

```diff
@@ -966,8 +967,16 @@ write_nm_conf_access_point(const NetplanNetDefinition* def, const char* rootdir,
         }
     }
 
-    /* NM connection files might contain secrets, and NM insists on tight permissions */
+    /* Create /run/NetworkManager/ with 755 permissions if the folder is missing.
+     * Letting the next invokation of safe_mkdir_p_dir do it would result in 
+     * more restrictive access because of the call to umask. */
+    nm_run_path = g_strjoin(G_DIR_SEPARATOR_S, rootdir ?: "", "run/NetworkManager/", NULL);
+    if (!g_file_test(nm_run_path, G_FILE_TEST_EXISTS))
```

Comment:
nitpick: you had some leading whitespace here, which doesn't match our code style, so I quickly fixed this up.